src/main/java/ru/curs/lyra/controller/LyraController.java

```diff
@@ -76,22 +80,51 @@ public ResponseEntity getData(@RequestParam Map<String, String> body) throws Exc
                 body.get("instanceId"),
                 clientParams
         );
-        String data = srv.getData(ctx, formInstantiationParameters, params).toString();
 
+        DataRetrievalParams dataRetrievalParams = new DataRetrievalParams();
+        dataRetrievalParams.setOffset(Integer.parseInt(body.get("offset")));
+        dataRetrievalParams.setLimit(Integer.parseInt(body.get("limit")));
+        dataRetrievalParams.setDgridOldPosition(Integer.parseInt(body.get("dgridOldPosition")));
+        dataRetrievalParams.setSortingOrFilteringChanged(Boolean.parseBoolean(body.get("sortingOrFilteringChanged")));
+        dataRetrievalParams.setFirstLoading(Boolean.parseBoolean(body.get("firstLoading")));
+        dataRetrievalParams.setRefreshId(getKeyValuesById(body.get("refreshId")));
+
+        JSONObject json = new JSONObject(body.get("context"));
+        String selectKey = ((JSONObject) json.get(REFRESH_PARAMS)).get(SELECT_KEY).toString();
+        dataRetrievalParams.setSelectKey(getKeyValuesById(selectKey));
+
+
+        //TODO: take into account user name here
+        //call context should be created with real user name
+        CallContext ctx = new SystemCallContext();
+
+        DataResult dataResult = srv.getData(ctx, formInstantiationParameters, dataRetrievalParams);
+
+        Object data = dataResult.getObjAddData() == null ? dataResult.getData() : dataResult.getObjAddData();
 
         HttpHeaders responseHeaders = new HttpHeaders();
         responseHeaders.set("Content-Type", "application/json;charset=UTF-8");
 
-        int totalCount = params.getTotalCount();
-        int firstIndex = params.getOffset();
-        int lastIndex = params.getOffset() + params.getLimit() - 1;
+        int totalCount = dataRetrievalParams.getTotalCount();
+        int firstIndex = dataRetrievalParams.getOffset();
+        int lastIndex = dataRetrievalParams.getOffset() + dataRetrievalParams.getLimit() - 1;
         responseHeaders.set("Content-Range", "items " + firstIndex + "-"
                 + lastIndex + "/" + totalCount);
 
-        ResponseEntity<String> responseEntity = new ResponseEntity<>(data, responseHeaders, HttpStatus.OK);
+        return new ResponseEntity<>(data, responseHeaders, HttpStatus.OK);
 
-        return responseEntity;
+    }
 
+    private Object[] getKeyValuesById(final String refreshId) {
```

Comment:
AFAIK this can be done in more elegant way using Jackson mapper, but it's ok for now.